src/components/organisms/FileCompareModal/FileCompareModal.tsx

```diff
@@ -0,0 +1,120 @@
+import {useEffect, useMemo, useState} from 'react';
+import {MonacoDiffEditor, monaco} from 'react-monaco-editor';
+import {useWindowSize} from 'react-use';
+
+import {Modal, Select} from 'antd';
+
+import fs from 'fs';
+import {join} from 'path';
+
+import {useAppDispatch, useAppSelector} from '@redux/hooks';
+import {closeFileCompareModal} from '@redux/reducers/ui';
+
+import {SelectItemImage} from '@components/atoms';
+
+import {useFileSelectOptions} from '@hooks/useFileSelectOptions';
+
+import {KUBESHOP_MONACO_THEME} from '@utils/monaco';
+
+import {ROOT_FILE_ENTRY} from '@shared/constants/fileEntry';
+
+import * as S from './FileCompareModal.styled';
+
+const HEIGHT_SIZE_PERCENTAGE = 0.85;
+const WIDTH_SIZE_PERCENTAGE = 0.92;
+
+const options: monaco.editor.IDiffEditorConstructionOptions = {
+  readOnly: true,
+  renderSideBySide: true,
+  inDiffEditor: true,
+  renderValidationDecorations: 'off',
+  minimap: {
+    enabled: false,
+  },
+};
+
+const FileCompareModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentFilePath = useAppSelector(state => state.ui.fileCompareModal.filePath);
+  const fileMap = useAppSelector(state => state.main.fileMap);
```

Comment:
`fileMap` is only used to check if a file exists, so we should make it a ref as we don't want the component to re-render when any other file changes:

```ts
const fileMapRef = useRefSelector(state => state.main.fileMap);
```